File: POO_Ads/Atividade 21-10-2024/atv_21-10-24/src/Audio_Exercise_Ely/ControleDeAudio.java
package Audio_Exercise_Ely;

public class ControleDeAudio {
    private int volume;

    // CRIANDO MEU CONSTRUTOR QUE INICIALIZA VOLUME
    public ControleDeAudio(int volume) {
        this.volume = volume;
    }

    // MÉTODO PARA AUMENTAR O VOLUME EM UM
    public int aumentaVolume() {
        if (volume < 10) {
            volume += 1;
        }
        return volume;
    }

    // MÉTODO PARA DIMINUIR VOLUME EM UM
    public int diminuiVolume() {
        if (volume > 0) {
            volume -= 1;
        }
        return volume;
    }

    // MÉTODO QUE RETORNA VOLUME
    public int lerVolume() {
        return volume;
    }
}

File: POO_Ads/Exercicio_2/src/Audio_Exercise_Ely/ControleDeAudio.ts
// CRIANDO A CLASSE CONTROLE DE ÁUDIO
class ControleDeAudio {
    private volume: number; // NUMBER SERVE TANTO PARA INT QUANTO PARA FLOAT, DIFERENTE DO JAVA(TEM INT E TEM FLOAT)

    // CRIANDO UM CONSTRUTOR, AQUI POSSO USAR DIRETO CONSTRUCTOR
    constructor(volume: number = 2){
        this.volume = volume; // MEU ATRIBUTO VOLUME VIRA O VOLUME PARÂMETRO
    }

    // CRIANDO OS MÉTODOS PARA AUMENTAR,DIMINUIR E LER VOLUME RESPECTIVAMENTE
    aumentar(): number{
        if(this.volume < 10){
            this.volume += 1;
        }
        return this.volume;
    }

    diminui(): number{
        if(this.volume > 0){
            this.volume -= 1;
        }
        return this.volume;
    }

    lerVolume(): number{
        return this.volume;
    }

    // AGORA, DEPOIS DE CRIAR OS MÉTODOS, CRIO A MAIN. ELA DIFERE APENAS UM POUCO DO JAVA, ASSIM COMO OS MÉTODOS
    public static main(): void {
        // NOTE QUE SE USA UM POUCO DE JAVASCRIPT NA DECLARAÇÃO DE UMA VARIÁVEL QUE CARREGA MEU CONSTRUTOR
        const audio = new ControleDeAudio(2);

        console.log(`Aumentando o audio: ${audio.aumentar()}`);
        console.log(`Aumentando o audio: ${audio.aumentar()}`);
        console.log(`diminuindo o audio: ${audio.diminui()}`);
        console.log(`lendo o audio: ${audio.lerVolume()}`);


    }

}
// CHAMA O MÉTODO PRINCIPAL
ControleDeAudio.main();